clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2277,6 +2294,12 @@ void CodeGenModule::SetLLVMFunctionAttributesForDefinition(const Decl *D,
 
   F->addFnAttrs(B);
 
+  if (getLangOpts().SYCLIsDevice && getCodeGenOpts().OptimizeSYCLFramework &&
```

Comment:
Why can't we re-use more of existing code and simply set `ShouldAddOptNone` to `false` under this condition?

Reply:
There is a problem that in case if we set `ShouldAddOptNone` to `false` then default code path adds `noinline` attribute what is undesirable.

https://github.com/intel/llvm/blob/sycl/clang/lib/CodeGen/CodeGenModule.cpp#L2232-L2236

Reply:
Got it, thanks